src/high-level/types/complex-double-float.lisp

```diff
@@ -0,0 +1,255 @@
+
+(defcompatible
+    (lambda (tensor)
+      (case (order tensor)
+        (1 '(vector/complex-double-float
+             tensor/complex-double-float))
+        (2 '(matrix/complex-double-float
+             tensor/complex-double-float))
+        (t '(tensor/complex-double-float))))
+  tensor/complex-double-float
+  matrix/complex-double-float
+  vector/complex-double-float)
+
+(defmethod dot ((vector1 vector/complex-double-float) (vector2 vector/complex-double-float))
+  (assert (cl:= (size vector1) (size vector2))
+          () "Vectors must have the same size. The first vector is size ~a and the second vector is size ~a."
+          (size vector1) (size vector2))
+  (loop :for i :below (size vector1)
+        :sum (* (tref vector1 i) (conjugate (tref vector2 i)))))
+
+(defmethod orthonormalize! ((m matrix/complex-double-float))
+  "Applies Gram-Schmidt to the columns of a full rank square matrix to produce a unitary matrix, replacing the elements"
+  (assert-square-matrix m)
+  ;; consider each column
+  (dotimes (j (ncols m))
+    ;; consider each preceding column, which together form an orthonormal set
+    (dotimes (jp j)
+      ;; compute the dot product of the columns...
+      (let ((scalar
+              (loop :for i :below (nrows m)
+                    :sum (* (tref m i j)
+                            (conjugate (tref m i jp))))))
+        ;; ... and do the subtraction.
+        (dotimes (i (nrows m))
+          (setf (tref m i j)
+                (cl:- (tref m i j)
+                      (* scalar
+                         (tref m i jp)))))))
+    ;; now j is orthogonal to the things that came before it. normalize it.
+    (let ((scalar
+            (sqrt
+             (loop :for i :below (nrows m)
+                   :sum (* (abs (tref m i j))
+                           (abs (tref m i j)))))))
+      (dotimes (i (nrows m))
+        (setf (tref m i j)
+              (/ (tref m i j) scalar)))))
+  m)
+
+(defmethod = ((tensor1 tensor/complex-double-float) (tensor2 tensor/complex-double-float) &optional (epsilon +double-comparison-threshold-strict+))
+  (unless (equal (shape tensor1) (shape tensor2))
+    (return-from = nil))
+  (map-indexes
+   (shape tensor1)
+   (lambda (&rest pos)
+     (unless (> epsilon
+                (abs (cl:- (apply #'tref tensor1 pos)
+                           (apply #'tref tensor2 pos))))
+       (return-from = nil))))
+  t)
+
+(defmethod = ((tensor1 matrix/complex-double-float) (tensor2 matrix/complex-double-float) &optional (epsilon +double-comparison-threshold-strict+))
+  (unless (equal (shape tensor1) (shape tensor2))
+    (return-from = nil))
+  (map-indexes
+   (shape tensor1)
+   (lambda (&rest pos)
+     (unless (> epsilon
+                (abs (cl:- (apply #'tref tensor1 pos)
+                           (apply #'tref tensor2 pos))))
+       (return-from = nil))))
+  t)
+
+;; ZUNCSD is broken in magicl lapack bindings
```

Comment:
Why? Is there an issue tracking this? If yes, add it here.

Reply:
Added. Now tracking in #72